### ISSUE TITLE
FF133. urlbar seems detached

### ISSUE THREAD

**Issue by x140x1n** [MrOtherGuy/firefox-csshacks#459]
using autohide_toolbox together with autohide_toolbox and upon opening firefox the url bar floats before hiding.
also the animation of the url bar on hover is not in sync with the rest of the toolbox.

https://github.com/user-attachments/assets/9133acbf-b9dd-41c3-813c-9fde06b700da

**Comment by MrOtherGuy** [comment]
Yeah, unfortunately now the urlbar has to be transformed separately, so yes it can appear "detached" from the rest of the toolbox. It shouldn't be super noticeable normally, but in some cases the movement can appear a bit out-of-sync from the actual toolbar. For startup I could probably just set it's opacity to 0 to make it invisible, but it can't actually start in "hidden" state for technical reasons.

**Comment by x140x1n** [comment]
I have fixed it for now inserting the following into autohide-toolbox.css.

below the first #navigator-toolbox {..} I added
```
#urlbar {
  opacity: 0;
}
```

and changed 

```
#mainPopupSet:has(> [role="group"][panelopen]) ~ toolbox #urlbar[popover],
#navigator-toolbox:is(:hover,:focus-within) #urlbar[popover],
#urlbar-container> #urlbar[popover]:is([focused],[open]){
  pointer-events: auto;
  opacity: 1;
  transition-delay: 33ms;
  transform: rotateX(0deg);
}
```

to include #urlbar after #urlbar-container

```
#mainPopupSet:has(> [role="group"][panelopen]) ~ toolbox #urlbar[popover],
#navigator-toolbox:is(:hover,:focus-within) #urlbar[popover],
#urlbar-container, #urlbar > #urlbar[popover]:is([focused],[open]){
  pointer-events: auto;
  opacity: 1;
  transition-delay: 33ms;
  transform: rotateX(0deg);
}
```
If there is a better way, which there probably is as I'm not a css expert, please let me know.

Thanks.

**Comment by x140x1n** [comment]
another issue I've noticed is when the ALT is pressed to show the menu bar the url bar doesn't jump down with the rest of the toolbox covering the menu, how can I fix that?

![menu](https://github.com/user-attachments/assets/cc554793-5a07-49a2-94cf-ef5e4f19f238)

**Comment by konsulas** [comment]
In autohide_bookmarks_and_main_toolbars.css urlbar also shows up in strange way.

**Comment by nerdsalot** [comment]
It worked. Thanks!
